src/lib/race.ts

```diff
@@ -242,3 +244,171 @@ export async function race(
 
   return API;
 }
+
+export function validateOptions(options: Partial<Options>): void {
```

Comment:
let's put this in "src/lib/options/validate-options.ts" to keep things clean here.